### DISCUSSION TITLE
arc without change in size?

### DISCUSSION THREAD

**Discussion by DOEHOONLEE** [ImageMagick/ImageMagick#4348] [Help]
Hi~!
I was having fun with arc where it makes an image into arch shape, but then it also changes its image width/height.

Is it possible to make an arc shape while maintaining its size?

Thank you!!

**Comment by fmw42** [comment]
What command? What version of Imagemagick? What platform/OS?

**Reply by DOEHOONLEE** [reply]
I am using version `7.1.0-6` on MacOS and following the example shown here.

![Screen Shot 2021-10-12 at 8 05 57 AM](https://user-images.githubusercontent.com/20305442/136865394-88523c29-ad0c-4a5e-a156-f98e48038b90.png)

I have this image `original.png` which is 1000x500 big.

![original](https://user-images.githubusercontent.com/20305442/136866154-ca89afb7-e16f-4017-b0cc-e9a0d94744ca.png)

Then I modify image with `convert original.png -distort Arc 120 original_arc.png`

![original_arc](https://user-images.githubusercontent.com/20305442/136866165-3e6d507d-7562-4e43-b50c-024e20670a0e.png)

I do get a nice arc shaped image, but then when I check its size, it is changed to 1262x615.

**Comment by fmw42** [comment]
-distort arc automatically expands the canvas to avoid cropping the image. You will have to resize afterwards back to the original size. Although the documentation says to use the viewport, it does not seem to work. See https://legacy.imagemagick.org/Usage/distorts/#distort_viewport

> Note that unlike the other image distortion operators, an 'Arc' distort will always set the size of the resulting image so that the complete source image is present. This includes any anti-aliasing edge pixels. As such the resulting image will rarely match the size of the input image.

> Only a the Viewport Distort Options, will allow you to change the resulting image size for a specific distortion.

**Reply by DOEHOONLEE** [reply]
Yeah, I actually tried that before posting, but then resize changes the shape's size as well like this.
`original_arc.png`
![original_arc](https://user-images.githubusercontent.com/20305442/136873131-9920cc61-672b-413a-9dc7-1594fb455aee.png)

`original_arc_resized.png`
![original_arc_resized](https://user-images.githubusercontent.com/20305442/136873137-b6fc64c8-8e58-4c77-ac1e-87b56a2487e0.png)

Let me try viewport distort option.

**Reply by fmw42** [reply]
The problem is that you have too much padding around it. Try trimming, then arcing, then resizing, then padding back

**Comment by snibgo** [comment]
Viewport does work, but in this case the viewport needs offsets.

However, as https://legacy.imagemagick.org/Usage/distorts/#distort_viewport says:

> It does not enlarge or scale the distorted image, just specify the location and area being viewed (the viewport) in the distorted image space.

I think the OP wants to scale the image. For that, as fmw42 says, we can "-resize" after the distortion. For example:
```
magick -size 1000x500 xc:#48a -virtual-pixel White -distort Arc 120 +repage -resize 1000x500 info:
```
The result is 1000x487 pixels. If desired, use "-repage 1000x500!" to make it exactly 1000x500.

**Reply by fmw42** [reply]
> Viewport does work, but in this case the viewport needs offsets.

You are right. I was dumbly thinking it would change the size. I should know better.

Adding to snibgo's answer, in IM 7, you can get the original size and store that in the same command line

`magick -size 1000x500 xc:"#48a" -virtual-pixel White -set option:dims "%wx%h\!" -distort Arc 120 +repage  -resize "%[dims]" -format "%wx%h\n" info:`

Returns:
1000x500

**Reply by DOEHOONLEE** [reply]
Hmm..
I tried to arc this image
![hamster](https://user-images.githubusercontent.com/20305442/136882720-a782ce5e-c7a1-4df5-9cbb-b3a5e6cb5254.png)
using
`convert hamster.png -virtual-pixel transparent -distort Arc 120 +repage -resize 500x500! hihi.png`

Then I got this.
![hihi](https://user-images.githubusercontent.com/20305442/136882739-0c8d54f2-aba3-4c2a-8593-dbd6d26fd630.png)

but can something like this be obtained?

![Screen Shot 2021-10-12 at 11 48 48 AM](https://user-images.githubusercontent.com/20305442/136882827-869d969c-7474-4308-aa49-f22d707dcf26.png)

Only the image gets arced, but draw it in some transparent canvas with specific size while maintaining the arced shape size?

**Reply by fmw42** [reply]
The arc process will enlarge the image. So if your specific size is not large enough, your resulting image will either be cropped or need to resized. What do you want to do in that case.  

To have transparency around the image, you will need to start with a transparent background. 

I suggest that you then trim the original, then arc that image to its bounds and then resize back to the trimmed size, then add your border back via -extent to achieve the final size that you want.

**Reply by DOEHOONLEE** [reply]
I found an interesting way of reshaping into arc.

![hamster](https://user-images.githubusercontent.com/20305442/136882720-a782ce5e-c7a1-4df5-9cbb-b3a5e6cb5254.png)

To reshape this image, I used this
![grad](https://user-images.githubusercontent.com/20305442/137242105-3123fffd-45be-494f-b82e-8c6b31c9a4de.png)

gradation image +

![Screen Shot 2021-10-14 at 11 47 20 AM](https://user-images.githubusercontent.com/20305442/137242169-a8450d4e-b116-4247-afb7-bc9b4f91cbd8.png)

displacement map to get arc shaped image :)
